clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -280,6 +280,7 @@ protected void onJoinPrepare(int generation, String memberId) {
         log.info("Revoking previously assigned partitions {} for group {}", subscriptions.assignedPartitions(), groupId);
         try {
             Set<TopicPartition> revoked = new HashSet<>(subscriptions.assignedPartitions());
+            revoked.addAll(subscriptions.pendingUnsubscribePartitions().keySet());
```

Comment:
It seems our latest agreement is that the partitions in pending unsubscribe state won't be passed to the rebalance listener. So should we leave this logic unchanged?